### DISCUSSION TITLE
Cannot install latest google provider along with kubernetes provider

### DISCUSSION THREAD

**Discussion by aru-trackunit** [apache/airflow#40236] [General]
### Apache Airflow Provider(s)

cncf-kubernetes, google

### Versions of Apache Airflow Providers

```
apache-airflow-providers-amazon==8.24.0
apache-airflow-providers-cncf-kubernetes==8.3.1
apache-airflow-providers-common-io==1.3.0
apache-airflow-providers-common-sql==1.11.1
apache-airflow-providers-databricks==6.5.0
apache-airflow-providers-ftp==3.7.0
apache-airflow-providers-github==2.6.1
apache-airflow-providers-google==10.19.0
apache-airflow-providers-hashicorp==3.7.1
apache-airflow-providers-http==4.10.0
apache-airflow-providers-imap==3.5.0
apache-airflow-providers-mysql==5.6.1
apache-airflow-providers-postgres==5.11.1
apache-airflow-providers-sftp==4.10.1
apache-airflow-providers-slack==8.7.1
apache-airflow-providers-smtp==1.6.1
apache-airflow-providers-snowflake==5.5.1
apache-airflow-providers-sqlite==3.7.1
apache-airflow-providers-ssh==3.11.1
```

### Apache Airflow version

2.8.4

### Operating System

Debian GNU/Linux 11 (bullseye)

### Deployment

Official Apache Airflow Helm Chart

### What happened

I am unable to install `apache-airflow-providers-google==10.19.0` along with `apache-airflow-providers-cncf-kubernetes==8.3.1` 

### What you think should happen instead

Packages should be both compatible

### How to reproduce

I couldn't build an image that contains `apache-airflow-providers-cncf-kubernetes==8.3.1` and `apache-airflow-providers-google==10.19.0` due to package conflicts.
I decided to downgrade cncf to version 7.13.0 that I had previously run the container and try to install newer kubernetes provider to see the exact problem:

`pip install apache-airflow-providers-cncf-kubernetes==8.3.1`
Output message:
```
Installing collected packages: cryptography, kubernetes_asyncio, kubernetes, apache-airflow-providers-cncf-kubernetes
  Attempting uninstall: cryptography
    Found existing installation: cryptography 38.0.4
    Uninstalling cryptography-38.0.4:
      Successfully uninstalled cryptography-38.0.4
  Attempting uninstall: kubernetes_asyncio
    Found existing installation: kubernetes-asyncio 24.2.3
    Uninstalling kubernetes-asyncio-24.2.3:
      Successfully uninstalled kubernetes-asyncio-24.2.3
  Attempting uninstall: kubernetes
    Found existing installation: kubernetes 23.6.0
    Uninstalling kubernetes-23.6.0:
      Successfully uninstalled kubernetes-23.6.0
  Attempting uninstall: apache-airflow-providers-cncf-kubernetes
    Found existing installation: apache-airflow-providers-cncf-kubernetes 7.13.0
    Uninstalling apache-airflow-providers-cncf-kubernetes-7.13.0:
      Successfully uninstalled apache-airflow-providers-cncf-kubernetes-7.13.0
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
gcloud-aio-auth 4.1.5 requires cryptography<39.0.0,>=2.0.0, but you have cryptography 42.0.8 which is incompatible.
pyopenssl 23.2.0 requires cryptography!=40.0.0,!=40.0.1,<42,>=38.0.0, but you have cryptography 42.0.8 which is incompatible.
Successfully installed apache-airflow-providers-cncf-kubernetes-8.3.1 cryptography-42.0.8 kubernetes-29.0.0 kubernetes_asyncio-29.0.0
```

Then I decided to install lower version of cryptography:
`pip install cryptography==38.0.4`

Output message:
```
Installing collected packages: cryptography
  Attempting uninstall: cryptography
    Found existing installation: cryptography 42.0.8
    Uninstalling cryptography-42.0.8:
      Successfully uninstalled cryptography-42.0.8
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
apache-airflow-providers-cncf-kubernetes 8.3.1 requires cryptography>=41.0.0, but you have cryptography 38.0.4 which is incompatible.
Successfully installed cryptography-38.0.4
```

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
Please do not use Github issues to discuss your troubleshooting. Use discussions instead if you need help.

Generally speaking it is up to you to resolve conflicts. Airflow provides you the constraints to install non-conflcting versions of dependencies for their latest releases, but if you want to mix and match different provider versions you are on your own to solve conflicts. For latest cncf + latest  google installed together - look at the latest constraints (either main or latest version released) and install cryptography version that is there. You will find information about constrainst in the "Installing from PyPI" instructions in our documentation.

**Comment by aru-trackunit** [comment]
Upgrading to airflow 2.9.2 solved the issue. However cncf + google were latest, airflow version was at 2.8.4.

Thanks for you time to look at this.
